lib/transports.js

```diff
@@ -1,14 +1,16 @@
 var events = require('events');
 var util = require('util');
+var assign = require('object-assign');
```

Comment:
Also, I apologize if that comes off as harsh. I just itemized the process I went through personally before I could understand the code I was looking at, which I don't feel is a very valuable addition to the project given the negative tradeoffs. I think a for loop is pretty simple and doesn't require an external dependency. If we could use `Object.assign()` in node, I'd :+1: it.